classes/class-wc-rest-connect-shipping-label-controller.php

```diff
@@ -51,10 +60,29 @@ private function validation_error( $fields ) {
 		);
 	}
 
+	private function starts_with( $haystack, $needle ) {
+		 return $needle === substr( $haystack, 0, strlen( $needle ) );
+	}
+
+	private function store_origin_address( $settings ) {
```

Comment:
This makes me a little nervous. A client could add arbitrary orig_* fields this way. Maybe we should have a whitelist instead?